modules/apps/27-interchain-accounts/host/keeper/relay_test.go

```diff
@@ -344,6 +344,20 @@ func (suite *KeeperTestSuite) TestOnRecvPacket() {
 			err := SetupICAPath(path, TestOwnerAddress)
 			suite.Require().NoError(err)
 
+			portId, err := types.GeneratePortID(TestOwnerAddress, ibctesting.FirstConnectionID, ibctesting.FirstConnectionID)
+			suite.Require().NoError(err)
+
+			// Get the address of the interchain account stored in state during handshake step
+			storedAddr, found := suite.chainB.GetSimApp().ICAHostKeeper.GetInterchainAccountAddress(suite.chainB.GetContext(), portId)
+			suite.Require().True(found)
+
+			icaAddr, err := sdk.AccAddressFromBech32(storedAddr)
+			suite.Require().NoError(err)
+
+			// Check if account is created
+			interchainAccount := suite.chainB.GetSimApp().AccountKeeper.GetAccount(suite.chainB.GetContext(), icaAddr)
```

Comment:
Is this only needed in `OnRecvPacket`? Should we be doing this elsewhere as well? If we already have an explicit test in `account_test.go` then is this neceessary to check in this test?

Reply:
This is the primary place that we're actually using the ICA and the critical flow to test IMO. It's not completely necessary as we have the explicit test in `account_test.go` but I think it's good to leave it in for sanity tbh.